pkg/ingester/activeseries/active_series.go

```diff
@@ -418,3 +504,60 @@ func resizeAndClear(l int, prev []uint32) []uint32 {
 	}
 	return p
 }
+
+type deletedSeries struct {
+	mu   sync.RWMutex
+	keys map[storage.SeriesRef]string
+	refs map[string]deletedSeriesRef
+
+	// buf can be used when write lock is held.
+	// If only read lock is held, then bufs pool should be used instead.
+	buf  []byte
+	bufs zeropool.Pool[[]byte]
+}
+
+type deletedSeriesRef struct {
+	ref storage.SeriesRef
+	key string
+}
+
+func (ds *deletedSeries) find(lbls labels.Labels) (deletedSeriesRef, bool) {
+	ds.mu.RLock()
+	defer ds.mu.RUnlock()
+
+	buf := ds.bufs.Get()
+	defer ds.bufs.Put(buf)
+
+	ref, ok := ds.refs[string(lbls.Bytes(buf))]
+	return ref, ok
```

Comment:
That's true but this code causes an allocation of the string, which is something I was trying to avoid. 
With `stringlabels` the `Bytes(buf)` call will be just copying the bytes from the internal datastructure to the buf (versus the slice implementation that has to build the bytes, but still no allocation)

How about I retrieve the buf from the pool before taking the mutex, but still do the `Bytes()` call in the map lookup?

Reply:
I've reordered to call the pool before taking the mutex, but left the `Bytes()` call inside, with a comment on why I'm doing that: https://github.com/grafana/mimir/pull/5678/commits/b16a9a4b91f0dd1109b1f0ce2ecabf08d3dd506b